app/src/main/java/com/example/palletify/ui/generator/GenerateWithImageScreen.kt

```diff
@@ -44,6 +64,53 @@ fun GenerateWithImageScreen(
     val paletteViewModel =
         ViewModelProvider(context as ViewModelStoreOwner).get(PaletteViewModel::class.java)
 
+    val colors = remember { mutableStateListOf<String>()}
+    val moreColors = remember { mutableStateListOf<String>()}
+
+    val colorPalette by generatorViewModel.colorPalette
+
+    var launchedEffectTriggered by remember { mutableStateOf(false) }
+
+    val imageUrl = uploadedImageUri.toString()
+
+    var selectedOption by remember {
+        mutableStateOf("")
+    }
+    val onSelectionChange = { text: String ->
+        selectedOption = text
+    }
+
+    LaunchedEffect(key1 = true) {
+        try {
+            val bitmap = convertImageUrlToBitmap(
+                imageUrl = imageUrl,
+                context = context
+            )
+            if (bitmap != null) {
+                launchedEffectTriggered = true
+                generatorViewModel.setColorPaletteFromImage(
+                    colors = extractColorsFromBitmap(
+                        bitmap = bitmap
+                    )
+                )
+                var colorCount = 0
+                colorPalette.forEach { (_, value) ->
+                    if (!(colors.contains(value) || moreColors.contains(value))) {
+                        if (colorCount < 5) {
+                            colors.add(value)
+                            colorCount++
+                        } else {
+                            moreColors.add(value)
+                        }
```

Comment:
i think in the main generator we set a max of 6 and a min of 3 to make it easier for ourselves

Reply:
gotcha i'll try to reuse those values